### PR TITLE
wip(gatsby-cli): add login, logout, whoami commands

### PR DESCRIPTION
**Pull request by gillkyle**
## Description

This adds the ability to log in to Gatsby Cloud from the CLI. The 3 following commands are hidden behind an experimental flag `GATSBY_EXPERIMENTAL_CLOUD_CLI` that when set enables the commands:

- `gatsby login`  - saves an access token after a login to the config store
- `gatsby logout` - removes the token
- `gatsby whoami` - hits the Gatsby Cloud API to return the name of the currentUser

Note: this gif is a little inaccurate as it shows a lot of debug level details like the polling in the CLI which has been removed. 

CleanShot 2020-11-23 at 17 11 13

The bajillion identical commits are me fighting TypeScript to pass our pre commit linting haha

_Good luck stealing my token you hackers because this is all on my local machine_ ;)

### CODE REVIEW

File: packages/gatsby-cli/src/login.ts [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+  await handleOpenBrowser(authUrl)
+
+  // Poll until the ticket has been verified, and should have the token attached
+  function pollForTicket(): Promise<ITicket> {
+    return new Promise(function (resolve): void {
+      // eslint-disable-next-line consistent-return
+      async function verify(): Promise<void> {
+        const ticket = await getTicket(ticketId)
+        if (ticket.verified) return resolve(ticket)
+        setTimeout(verify, 3000)
```
Comment (mfrachet) [inline packages/gatsby-cli/src/login.ts:90]: Nitpick: we could save an eventual last hit on the API by clearing the setTimeout ID when resolving the ticket

File: packages/gatsby-cli/src/util/manage-token.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+  if (tokenHasExpired) {
+    report.warn(`Your token has expired, you may need to login again`)
+  }
+  return getConfigStore().get(tokenKey)
+}
+
+export const setToken = (token: string | null, expiration: string): void => {
+  getConfigStore().set(tokenKey, token)
+  // we would be able to decode an expiration off the JWT, but the auth service isn't set up to attach it to the token
+  getConfigStore().set(tokenExpirationKey, expiration)
```
Comment (mfrachet) [inline packages/gatsby-cli/src/util/manage-token.ts:20]: Very nitpicky:

```js
const store = getConfigStore()

store.set(tokenKey, token)
store.set(tokenExpirationKey, expiration)
```

**Review by mfrachet — APPROVED** [review]
I like your coding style, code is self-explanatory and easy to figure out 👍🏻 . I'm nit picky on tiny things but feel free to ignore them if they don't make sense :)

**Review by mxstbr — DISMISSED** [review]
This looks fantastic! :100:

**Pull request merged**